jdaviz/app.py

```diff
@@ -1244,11 +1230,27 @@ def _on_data_deleted(self, msg):
                 self.state.data_items.remove(data_item)
 
     @staticmethod
-    def _create_data_item(label):
+    def _create_data_item(data):
+        ndims = len(data.shape)
+        component_ids = [str(c) for c in data.component_ids()]
+        if 'Identifier' in component_ids:
+            typ = 'table'
+        elif ndims == 1:
+            typ = '1d spectrum'
+        elif ndims == 2:
+            typ = '2d spectrum' if 'Wavelength' in component_ids else 'image'
+        elif ndims == 3:
+            typ = 'cube'
+        else:
+            typ = 'unknown'
```

Comment:
reviewers (cc @pllim): please try to break this logic and preferably come up with a more robust solution

Reply:
If we want to stick with [APE 14](https://github.com/astropy/astropy-APEs/blob/main/APE14.rst), Stuart Mumford said our best bet is the `world_axis_physical_types` attribute values.

For an image data, `Data.coords.world_axis_physical_types` gave me `['pos.eq.ra', 'pos.eq.dec']`, while the example Mosviz notebook 2d spectrum gave me `['em.wl', None]`. Full list of possible values appear to be listed at https://github.com/astropy/astropy/blob/main/astropy/wcs/wcsapi/data/ucds.txt . Do we want to go this route?